driver/src/test/java/org/neo4j/driver/internal/async/pool/NettyChannelHealthCheckerTest.java

```diff
@@ -91,6 +97,55 @@ void shouldAllowVeryOldChannelsWhenMaxLifetimeDisabled()
         assertThat( await( healthy ), is( true ) );
     }
 
+    @Test
+    void shouldFailAllConnectionsCreatedOnOrBeforeExpirationTimestamp()
+    {
+        PoolSettings settings = new PoolSettings( DEFAULT_MAX_CONNECTION_POOL_SIZE,
+                                                  DEFAULT_CONNECTION_ACQUISITION_TIMEOUT, NOT_CONFIGURED, DEFAULT_IDLE_TIME_BEFORE_CONNECTION_TEST );
+        Clock clock = Clock.SYSTEM;
+        NettyChannelHealthChecker healthChecker = newHealthChecker( settings, clock );
+
+        long initialTimestamp = clock.millis();
+        List<Channel> channels = IntStream.range( 0, 100 ).mapToObj( i ->
+                                                                     {
+                                                                         Channel channel = new EmbeddedChannel();
+                                                                         setCreationTimestamp( channel, initialTimestamp + i );
+                                                                         return channel;
+                                                                     } ).collect( Collectors.toList() );
+
+        int authorizationExpiredChannelIndex = channels.size() / 2 - 1;
+        healthChecker.onExpired( new AuthorizationExpiredException( "", "" ), channels.get( authorizationExpiredChannelIndex ) );
+
+        for ( int i = 0; i < channels.size(); i++ )
+        {
+            Channel channel = channels.get( i );
+            boolean health = Objects.requireNonNull( await( healthChecker.isHealthy( channel ) ) );
+            boolean expectedHealth = i > authorizationExpiredChannelIndex;
+            assertEquals( expectedHealth, health, String.format( "Channel %d has failed the check", i ) );
+        }
+    }
+
```

Comment:
Maybe a test that checks newer connections are still healthy?

Reply:
It actually checks both, see the `expectedHealth` variable definition - it depends on the position in the list, which is in chronological order.

Reply:
ah of course, I missread it